### ISSUE TITLE
Canvas text scaling on Linux w/ GTK

### ISSUE THREAD

**Issue by cyqsimon** [CelestiaProject/Celestia#1192]
I recently installed Celestia on Manjaro from the official repository.

While all the GTK elements are scaled correctly, the rendered texts on the main canvas were not. On my 4K screen they are barely visible. See screenshot:

![shot](https://user-images.githubusercontent.com/28627918/144102521-e7b5c3b1-4ba2-4b25-b274-a88082451094.png)

Is there an option to set text scaling for the canvas? Alternatively, is there a way to use a larger font size?

The installed Celestia version is 1.6.2.2, in case that's relevant.

**Comment by 375gnu** [comment]
@cyqsimon unfortunately version 1.6 uses only raster fonts in TXF format and it ships only a few selected sizes. Now you have 2 options: 1) compile Celestia 1.7 which works with scalable TTF fonts, 2) try to find utility which converts TTD into TXF, our original one is lost. The edit `celestia.cfg` file in Celestia installation directory (or `~/.celestia.cfg`) and amend `Font`, `LabelFont` and `TitleFont` options.

**Comment by 375gnu** [comment]
Actually version 1.7 works on high dpi screens much better.

**Comment by munix9** [comment]
Maybe you could also try the AppImage: https://github.com/CelestiaProject/Celestia/issues/333

**Comment by cyqsimon** [comment]
Yeah txf seems to be a really obscure format. I wasn't able to find the source for `ttf2txf` after extensive scouring (I suspect many have already tried and failed lol).

For now I followed your advice and set `Font "sans20.txf"` and `LabelFont "sans20.txf"` in `/usr/share/celestia/celestia.cfg` as a stopgap measure, which slightly improved the situation. Much thanks for the help.

I'm not really in any kind of hurry so I'm fine waiting for 1.7 release. Is there any rough estimate for its release date?

**Comment by 375gnu** [comment]
no, we don't have any release date for 1.7. so Appimage would be better solution if you don't want to compile yourself.

**Comment by 375gnu** [comment]
but if you want to use 1.6.2, then you can clone https://gitlab.com/HaydenKow/tt2txf then edit makefile.txt, serch for `all:` and leave in this line only `ttf2txf`: `all: ttf2txf`. Then run `mkdir obj && make -f makefile.txt`. And you get `tt2txf`

**Comment by cyqsimon** [comment]
Thanks for the link! The tool compiled and worked great. I generated a bunch of sizes and ended up using 36pts and 60pts for text and title respectively. Much better now:

![After](https://user-images.githubusercontent.com/28627918/144282823-b278b3e8-92ad-479f-b5da-6e37fb1aeae0.png)

The line spacing still looks a bit crooked but that's not a big deal TBH.

Here's all the TXF files I generated in case anyone else wants to use them.

[notosans-txf_12-128.tar.gz](https://github.com/CelestiaProject/Celestia/files/7635988/notosans-txf_12-128.tar.gz)
